### ISSUE TITLE
Troubleshooting doc out of date?

### ISSUE THREAD

**Issue by pfernandez** [clojure-lsp/clojure-lsp#1341]
**Describe the bug**
When trying to follow the troubleshooting steps at https://clojure-lsp.io/troubleshooting/#server-is-not-initializing, no error message is printed to stdout when running
```
$ clojure-lsp
{}

```
as instructed. It's I've tried this in MacOS and Parallels (Debian) with the same result.

**To Reproduce**
Try running `{}\n\n` with the latest version of clojure-lsp.

**Comment by pfernandez** [comment]
FYI, the thing I was troubleshooting was that clojure-lsp isn't initializing in the last release of coc.nvim (at least for me). I've opened an issue with them as well: https://github.com/neoclide/coc.nvim/issues/4329.

**Comment by mainej** [comment]
@pfernandez see #1342 for a fix to `clojure-lsp`'s troubleshooting docs.

I'm not super familiar with coc.nvim, but scanning through https://github.com/neoclide/coc.nvim/issues/4329, here's my guess as to what's happening:

In [a3d271d](https://github.com/neoclide/coc.nvim/commit/a3d271d), coc.nvim completely refactored how it reads and generates its configuration settings. I would guess that some of these settings are fed into the "initialization options" that the client (coc.nvim) sends to the server (clojure-lsp) during the `initialize` request. From the log in your issue it appears that the server is responding with -32603, a code that means the server threw an exception during that request.

So it's quite possible that there's a bug on both sides... coc.nvim is sending new or different params that are malformed in some way, and clojure-lsp isn't careful enough about how it parses or uses those params.

The server-side exception stacktrace might give a hint as to what parameter is causing a problem. That stacktrace should be logged, so check the [troubleshooting docs](https://clojure-lsp.io/troubleshooting/#server-log) about how to inspect those logs. Unfortunately in the very early stages of startup, logs sometimes go to a different file. So if you're not seeing a stacktrace in the logs, we might have to go hunting.

**Comment by pfernandez** [comment]
Thanks! I tried it and it works now. I did leave a comment about the `--trace-level` part though. Maybe that's going to be added in the next release.

Honestly I've spent all the time I have at the moment, and I'm hoping these two issues I've created will provide enough breadcrumbs that the CoC folks will figure it out. Or maybe @NoahTheDuke, the author of the [coc-clojure](https://github.com/NoahTheDuke/coc-clojure) LSP wrapper. 😄 If not I'll try to circle back at some point.

**Comment by NoahTheDuke** [comment]
I don't have much dev time on the weekends but I'll take a look when I get a chance. I try to keep up to date with coc's changes, but tbh, they move so fast it can be tiring to keep up. Maybe I should just stick with supporting major releases lol.

**Comment by chemzqm** [comment]
The issue caused by `"clojure.initialization-options.log-path": ""` coc-clojure could set the default value to `null` or exclude it if it's empty string.

**Comment by NoahTheDuke** [comment]
Thanks @chemzqm! I'll fix and release tonight.

**Comment by pfernandez** [comment]
I just tried it all in both MacOS and a Debian VM, and it's working great now. :100: 

While you're all in the room-- thanks for making Vim the most awesome IDE ever, especially for Clojure.

An nrepl isn't even necessary anymore with such great static analysis. Pairing it with [zepl.vim](https://github.com/axvr/zepl.vim) to `ReplSend` expressions to a basic repl, I'm pretty sure I've got every feature that VSCode has with Calva.
